### PR TITLE
added request param to get_url

### PR DESCRIPTION
**Pull request by lwyszomi**
discussion: http://manage.dimagi.com/default.asp?154287#859454

### CODE REVIEW

**Comment by dimagimon** [comment]
:+1: No lint errors found.

**Comment by lwyszomi** [comment]
@czue

File: corehq/apps/reports/dispatcher.py [hunk 1]
```diff
@@ -184,7 +184,7 @@ def navigation_sections(cls, context):
                     else:
                         report_contexts.append({
                             'is_active': report.slug == current_slug,
-                            'url': report.get_url(domain=domain),
+                            'url': report.get_url(domain=domain, request=request),
```
Comment (czue) [inline corehq/apps/reports/dispatcher.py]: it looks like there are at least some references to `get_url` that aren't prepared to take an extra argument. for example here: https://github.com/dimagi/commcare-hq/blob/master/corehq/__init__.py#L155

@dimagi/team-commcare-hq does anyone have any opinions on this? (some custom reports want to override `get_url` with user context-specific information). i'm not a huge fan of adding more stuff to the api, but not sure i have a better proposal

Comment (esoergel) [inline corehq/apps/reports/dispatcher.py]: Here are the current implementations of get_url:
```
$ ag "def get_url\(" .
corehq/apps/dashboard/models.py
104:    def get_url(self, request):

corehq/apps/receiverwrapper/models.py
251:    def get_url(self, repeate_record):
272:    def get_url(self, repeat_record):

corehq/apps/adm/reports/__init__.py
47:    def get_url(cls, domain=None, render_as=None, **kwargs):

corehq/apps/reports/generic.py
635:    def get_url(cls, domain=None, render_as=None, **kwargs):

corehq/__init__.py
155:            def get_url(cls, domain):

custom/_legacy/pact/management/commands/__init__.py
35:    def get_url(self, url, retry=0):
```
All the standard reports use the generic `get_url` unmodified, it seems, beyond that, there's not really any agreement on arguments.  Passing request in seems reasonable, as it contains a great deal of context.  Does anything besides standard and custom reports use this dispatcher?  (Do user defined report?).

Comment (czue) [inline corehq/apps/reports/dispatcher.py]: @lwyszomi see above

Comment (czue) [inline corehq/apps/reports/dispatcher.py]: (at a minimum I think we need to fix / test the reference i mentioned)

**Comment by dimagimon** [comment]
:+1: No lint errors found.

**Comment by lwyszomi** [comment]
@czue 

Updated

**Pull request merged**